Introduce and Use NetworkIndicator component

jarolrod opened the pull request:
Signals to the user when they are running on a network other than main.

Closes https://github.com/bitcoin-core/gui-qml/issues/233

#### Light
| testnet | signet | regtest |
| ------- | ------ | ------- |
| <img width="752" alt="Screen Shot 2023-02-08 at 6 48 47 PM" src="https://user-images.githubusercontent.com/23396902/217695964-9ceaea19-461c-485a-b41e-b3d0d3ff4211.png"> |  <img width="752" alt="Screen Shot 2023-02-08 at 6 39 00 PM" src="https://user-images.githubusercontent.com/23396902/217695987-07455475-ede9-4394-bdb5-14e0a8246123.png"> | <img width="752" alt="Screen Shot 2023-02-08 at 6 39 21 PM" src="https://user-images.githubusercontent.com/23396902/217696012-a2dda1e0-e7a1-4ecc-9655-2a7ea55c9f75.png"> |



#### Dark
| testnet | signet | regtest |
| ------- | ------ | ------- |
| <img width="752" alt="Screen Shot 2023-02-07 at 5 24 23 PM" src="https://user-images.githubusercontent.com/23396902/217380996-91c16d78-fc49-49b5-9e33-73708dce3c15.png"> | <img width="752" alt="Screen Shot 2023-02-07 at 5 24 50 PM" src="https://user-images.githubusercontent.com/23396902/217381059-3147923e-b73f-4e2f-935d-77ce74554b71.png"> | <img width="752" alt="Screen Shot 2023-02-07 at 5 25 34 PM" src="https://user-images.githubusercontent.com/23396902/217381100-9f1c67e8-70e5-4658-ac73-7a78d6d26339.png"> |


[![Windows](https://img.shields.io/badge/OS-Windows-green)](https://api.cirrus-ci.com/v1/artifact/github/bitcoin-core/gui-qml/win64/insecure_win_gui.zip?branch=pull/<PR>)
[![Intel macOS](https://img.shields.io/badge/OS-Intel%20macOS-green)](https://api.cirrus-ci.com/v1/artifact/github/bitcoin-core/gui-qml/macos/insecure_mac_gui.zip?branch=pull/<PR>)
[![Apple Silicon macOS](https://img.shields.io/badge/OS-Apple%20Silicon%20macOS-green)](https://api.cirrus-ci.com/v1/artifact/github/bitcoin-core/gui-qml/macos_arm64/insecure_mac_arm64_gui.zip?branch=pull/<PR>)
[![ARM64 Android](https://img.shields.io/badge/OS-Android-green)](https://api.cirrus-ci.com/v1/artifact/github/bitcoin-core/gui-qml/android/insecure_android_apk.zip?branch=pull/<PR>)

jarolrod commented:
Updated from [bcb3aee ](https://github.com/bitcoin-core/gui-qml/commit/bcb3aeefeb89f58b279776a1cb5803b547152624)to [dd76b6e](https://github.com/bitcoin-core/gui-qml/commit/dd76b6ecced71cf3b48adf76265769258277b5e0), [compare](https://github.com/bitcoin-core/gui-qml/compare/bcb3aeefeb89f58b279776a1cb5803b547152624..dd76b6ecced71cf3b48adf76265769258277b5e0)

- Changes: I've been informed by @GBKS that the text should always be white

hebasto commented:
Rebase?

johnny9 commented:
ACK 932622e674b6e2476f635e901b2b00cbc58b7e6a

Changes look good and match figma spec.

![Screenshot from 2023-02-08 22-17-32](https://user-images.githubusercontent.com/985648/217855360-be746dc0-a602-4ece-82db-1f681c8b9afb.png)
![Screenshot from 2023-02-08 22-17-10](https://user-images.githubusercontent.com/985648/217855364-97952a8a-c9af-4574-af78-69854e581b4c.png)

johnny9 reviewed (approved):
ACK 932622e674b6e2476f635e901b2b00cbc58b7e6a

Changes look good, matches the figma spec, and no errors.

![Screenshot from 2023-02-08 22-17-32](https://user-images.githubusercontent.com/985648/217856104-02fcb823-e4dc-4f85-9d25-bd045cf9d1a2.png)
![Screenshot from 2023-02-08 22-17-10](https://user-images.githubusercontent.com/985648/217856105-58ef02f4-a6bb-4617-b71e-38c6495b822b.png)

johnny9 commented on src/qml/pages/node/NodeRunner.qml:
```diff
@@ -18,7 +18,15 @@ Page {
+        spacing: 30
         anchors.centerIn: parent
+        BlockClock {
+          id: blockClock
```
Whitespace issue here

jarolrod replied:
thanks, fixed

johnny9 reviewed (changes requested):
minor whitespace issue.

jarolrod commented:
Updated from [932622e ](https://github.com/bitcoin-core/gui-qml/commit/932622e674b6e2476f635e901b2b00cbc58b7e6a)to [01b3909](https://github.com/bitcoin-core/gui-qml/commit/01b3909f611b3a1db4d1d827350ab2b72c97e2f6), [compare](https://github.com/bitcoin-core/gui-qml/compare/932622e674b6e2476f635e901b2b00cbc58b7e6a..01b3909f611b3a1db4d1d827350ab2b72c97e2f6)

changes: addressed indenting issue but also removed the id for the block clock component since its not being used here.

johnny9 reviewed (approved):
ACK 01b3909f611b3a1db4d1d827350ab2b72c97e2f6

Changes look good, no errors, and matches the figma

![Screenshot from 2023-02-09 11-53-42](https://user-images.githubusercontent.com/985648/217920250-088392c3-77c4-477f-b344-aeffce61f6da.png)
![Screenshot from 2023-02-09 11-54-05](https://user-images.githubusercontent.com/985648/217920253-1108ea01-008b-4dda-8f6a-f8430864c5f3.png)

GBKS commented:
Sorry for the late feedback. A couple of details don't match the design. The font should be 15px, and vertical spacing should be 2 (horizontal remains 7). As is right now, the indicator looks slightly too big compared to everything else.
